tests/test_data_sources.py

```
@@ -176,3 +175,17 @@ def test_arrow_datasource_multiple_files(spark):
         rows = df.collect()
         names = {row["name"] for row in rows}
         assert names == {"Alice", "Bob", "Charlie", "Diana"}
+
+def test_jsonplaceholder_posts(spark):
+     spark.dataSource.register(JSONPlaceholderDataSource)
```

Ruff (0.12.2)

`JSONPlaceholderDataSource` may be undefined, or defined from star imports

(F405)